kernel/src/common/result.rs

```diff
@@ -0,0 +1,18 @@
+//! result.rs -- Standard return type for invoking operations, returning
+//! success or an error code.
+//!
+//!  Author: Philip Levis <pal@cs.stanford.edu>
+//!  Date: Nov 21, 2016
+
+enum Result {
+    SUCCESS,
+    FAIL, // Generic failure condition
+    EBUSY, // Underlying system is busy; retry
+    EALREADY, // The state requested is already set
```

Comment:
The state transition signals an event to denote it has completed. E.g., turning something on has an event that denotes it's on. If you return SUCCESS, that implies an event will be signaled later. If the device is already on, then signaling the event requires having an interrupt source (invoking the callback from within the call isn't what the caller expects). There are therefore cases where the device needs to be a timer client in order to meet the interface.

That being said, I agree that this is an edge case very far from what we've dealing with right now, so I'm fine leaving it out.